addons/binding/org.openhab.binding.wakeonlan/src/main/java/org/openhab/binding/wakeonlan/internal/SmarthomeCommandHelperImpl.java

```diff
@@ -0,0 +1,362 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * @author Ganesh Ingle <ganesh.ingle@asvilabs.com>
+ */
+
+package org.openhab.binding.wakeonlan.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.items.ItemNotFoundException;
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.items.events.ItemEventFactory;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.TypeParser;
+import org.openhab.binding.wakeonlan.SmarthomeCommandHelper;
+import org.slf4j.Logger;
+
+/**
+ * Helper class to send one or more commands to openhab items or channels
+ *
+ * @author Ganesh Ingle - Initial contribution
+ *
+ */
+// @Component
+public class SmarthomeCommandHelperImpl implements SmarthomeCommandHelper {
+
+    private static Map<String, Long> lastCommandTimeByItem = new HashMap<String, Long>();
+    public static final int COMMAND_DELAY_MS = 200; // Minimum delay between commands to same target
+    // protected WakeOnLanHandlerFactory factory = null;
+    // @Reference(policy = ReferencePolicy.DYNAMIC)
+    protected volatile ThingRegistry thingRegistry;
+    // @Reference(policy = ReferencePolicy.DYNAMIC)
+    protected volatile ItemRegistry itemRegistry;
+    // @Reference(policy = ReferencePolicy.DYNAMIC)
+    protected volatile ItemChannelLinkRegistry itemChannelLinkRegistry;
+    // @Reference(policy = ReferencePolicy.DYNAMIC)
+    protected volatile EventPublisher eventPublisher;
```

Comment:
The idea here was to imagine a PC as ON/OFF device. If you have different commands at different places, wakeup/ regular ping / shutdown , user gets lost various tabs. In fact many users have hand coded their custom item / layouts, just to get ON/OFF on a single switch item. Couple of users did ask for such a feature from binding itself, those who didn't have expertise around config files.

Reply:
Regarding shutdown command delegation to other items / channels (e.g Kodi binding supports shutdown via its own protocol), I agree that functionality should somehow be wrapped in other core level extension. Need advise here. I just didn't find any other way to invoke features provided by other bindings. And I didn't like the rules/config file options from OH1 philosophy. OH2 is supposed to be mostly Linux free, coding free, command line free. :)

Reply:
Thanks for your views on this @asvilabs! I just want to make sure we are headed in the right direction before spending time on reviewing/reworking the code. I'm eager to review it since the WoL binding is one of the last OH1 bindings I'm using. :-)

Reply:
@wborn I am afraid you will have to keep using the WoL 1.x binding for longer :-(.
But the code is still available here, published under the EPL; so if you feel like it, you could start a new PR and use this work as a basis for it, wdyt?

Reply:
I've created https://github.com/openhab/openhab2-addons/issues/3799 for continuing the discussion.